src/commands/doctor.rs

```diff
@@ -0,0 +1,115 @@
+use std::{fmt::Debug, path::PathBuf};
+
+use anyhow::Result;
+use clap::Parser;
+use dialoguer::{console::Emoji, Confirm, Select};
+use futures::FutureExt;
+use spin_doctor::{Diagnosis, DryRunNotSupported};
+
+#[derive(Parser, Debug)]
+#[clap(hide = true, about = "Detect and fix problems with Spin applications")]
+pub struct DoctorCommand {
+    #[clap(short = 'f', long, default_value = "spin.toml")]
+    file: PathBuf,
+}
+
+impl DoctorCommand {
+    pub async fn run(self) -> Result<()> {
+        println!("{icon}The Spin Doctor is in.", icon = Emoji("📟 ", ""));
+        println!(
+            "{icon}Checking {}...",
+            self.file.display(),
+            icon = Emoji("🩺 ", "")
+        );
+
+        let count = spin_doctor::Checkup::new(self.file)
+            .for_each_diagnosis(move |diagnosis, patient| {
+                async move {
+                    show_diagnosis(&*diagnosis);
+
+                    if let Some(treatment) = diagnosis.treatment() {
+                        let dry_run = match treatment.dry_run(patient).await {
+                            Ok(desc) => Some(desc),
+                            Err(err) => {
+                                if !err.is::<DryRunNotSupported>() {
+                                    show_error("Treatment dry run failed: ", err);
+                                }
+                                return Ok(());
+                            }
+                        };
+
+                        let should_treat = prompt_treatment(treatment.summary(), dry_run)
+                            .unwrap_or_else(|err| {
+                                show_error("Prompt error: ", err);
+                                false
+                            });
+
+                        if should_treat {
+                            match treatment.treat(patient).await {
+                                Ok(()) => {
+                                    println!("{icon}Treatment applied!", icon = Emoji("❤️ ", ""));
```

Comment:
Something goes slightly awry in the alignment of this line.  It's as if the space is being ignored, even though it's definitely there!

![image](https://user-images.githubusercontent.com/865538/236734910-1d66106d-5d28-488f-8d9a-78ce64451233.png)

Reply:
Hmm yes I can reproduce this problem with certain glyphs in some terminals (fonts?). It seems like certain double-width emoji are treated as single-width...

Reply:
Seems to be an issue with certain unicode emoji that are encoded as a "legacy" single-codepoint graphical character plus the "emoji presentation selector". Oh unicode :roll_eyes:

Reply:
I think I have this fixed now, though we should keep this in mind when we decide on specific emoji / whether to have them at all.